cmd/test-server/kcp/shard.go

```diff
@@ -143,26 +164,32 @@ func Start(ctx context.Context, name, runtimeDir, logFilePath string, args []str
 	}
 	logger.Info("Found kubeconfig", "path", kubeconfigPath)
 
+	return nil
+}
+
+func (s *Shard) WaitForReady(ctx context.Context) (<-chan error, error) {
 	// wait for readiness
+	logger := klog.FromContext(ctx)
 	logger.Info("Waiting for shard /readyz to succeed")
 	for {
 		time.Sleep(time.Second)
 
 		select {
 		case <-ctx.Done():
 			return nil, fmt.Errorf("context canceled")
-		case err := <-terminatedCh:
+		case err := <-s.terminatedCh:
 			var exitErr *exec.ExitError
 			if err == nil {
-				return nil, fmt.Errorf("kcp shard %s terminated unexpectedly with exit code 0", name)
+				return nil, fmt.Errorf("kcp Shard %s terminated unexpectedly with exit code 0", s.name)
 			} else if errors.As(err, &exitErr) {
-				return nil, fmt.Errorf("kcp shard %s terminated with exit code %d", name, exitErr.ExitCode())
+				return nil, fmt.Errorf("kcp Shard %s terminated with exit code %d", s.name, exitErr.ExitCode())
 			}
-			return nil, fmt.Errorf("kcp shard %s terminated with unknown error: %w", name, err)
+			return nil, fmt.Errorf("kcp Shard %s terminated with unknown error: %w", s.name, err)
 		default:
 		}
 
 		// intentionally load again every iteration because it can change
+		kubeconfigPath := filepath.Join(s.runtimeDir, "admin.kubeconfig")
 		configLoader := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfigPath},
 			&clientcmd.ConfigOverrides{CurrentContext: "system:admin"},
```

Comment:
okay, I think it works because the rest client get an absolute path (we don't append `clusters/root`)

Reply:
I think it works because this is a valid context for the shard admin.kubeconfig (it's the shard-admin user), but it's not for the front-proxy one?

```
$ grep system:admin .kcp/admin.kubeconfig 
$ grep -B3 system:admin .kcp-0/admin.kubeconfig 
- context:
    cluster: base
    user: shard-admin
  name: system:admin
```